flipperdevices/flipperzero-firmware
# NFC: Fix MIFARE DESfire info action to open app menu

When examining a MIFARE DESfire card, selecting "More >" currently leads to the file menu for a single application even if the scanned card contains multiple applications.

On examining the source code, a MIFARE DESfire application selection menu is present as `NfcSceneMfDesfireData`. This change updates the MIFARE DESfire Info "More >" action to open the application selection menu. That menu may then be used to open the file selection menu for any application in the read MIFARE DESfire card data.

Tested interactively with
https://github.com/flipperdevices/flipperzero-sd-card-examples/blob/c4cbdcd94706836fe0e3cda42587814237b1cfde/nfc/Desfire.nfc

Existing MIFARE DESfire menu flow:

![desfire-1](https://user-images.githubusercontent.com/2831985/204123367-99fe51ae-c5b0-4568-912e-8b37908ad091.png)
![desfire-3](https://user-images.githubusercontent.com/2831985/204123371-aa9ba146-8e11-48ab-ae31-6177aad54bda.png)

New MIFARE DESfire menu flow:

![desfire-1](https://user-images.githubusercontent.com/2831985/204123367-99fe51ae-c5b0-4568-912e-8b37908ad091.png)
![desfire-2](https://user-images.githubusercontent.com/2831985/204123379-3c455d3a-754d-476d-889c-4db8be179317.png)
![desfire-3](https://user-images.githubusercontent.com/2831985/204123371-aa9ba146-8e11-48ab-ae31-6177aad54bda.png)

# What's new

See above

# Verification 

* Copy [Desfire.nfc](https://github.com/flipperdevices/flipperzero-sd-card-examples/blob/c4cbdcd94706836fe0e3cda42587814237b1cfde/nfc/Desfire.nfc) to Flipper Zero SD card
* Open NFC app, select "Saved," and select copied `Desfire.nfc` file
* Select "Info"
* The MIFARE DESfire summary info screen is now displayed. Press `>` to select `More >`.
  * Existing behavior: File selection menu appears
  * Updated behavior: MIFARE DESfire application selection menu appears.

# Checklist (For Reviewer)
